sklearn/feature_extraction/text.py

```diff
@@ -681,6 +683,30 @@ def fit(self, X, y=None):
 
         self._validate_params()
 
+        if self.analyzer != 'word' and self.stop_words is not None:
```

Comment:
Can this just happen in _build_analyzer to avoid code duplication?

Reply:
@jnothman  -I have moved the analyzer related warnings inside the build_analyzer . Can  you please check if this is what you had in mind . Also , the  unit test for HashVectorizer fails   with this approach . Need to find the root cause for that .